source/agora/common/Hash.d

```diff
@@ -0,0 +1,50 @@
+/*******************************************************************************
+
+    Contains hashing functions
+
+    Copyright:
+        Copyright (c) 2019 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.common.Hash;
+
+import std.algorithm;
+import std.digest.sha;
+import std.range;
+import std.stdio;
+import std.string;
+import std.utf;
+
+import core.stdc.stdlib;
+
+import geod24.bitblob;
+
+/// 256 bits opaque blob, used for hashs
+public alias Hash = BitBlob!256;
+
+/// Simple wrapper around sha256 which returns Hash
+public Hash hash (T) (auto ref T arg)
+{
+    static ubyte[] buffer;
+    buffer.length = 0;
+    assumeSafeAppend(buffer);
+    scope void delegate(scope const ubyte[] data) sink =
+                   (scope data) { buffer ~= data; };
+
+    import agora.binding.types.Utils;
+    static if (is(T : E[], E))
+        const bs = ByteSlice.make(arg);
+    else static if (is(typeof(arg.serialize(sink))))
+    {
+        arg.serialize(sink);
+        const bs = ByteSlice.make(buffer);
+    }
+    else
+        const bs = ByteSlice.make(XDRToOpaque(arg));
+    return sha256(bs);
```

Comment:
Is that ?... I mean ? I'm scared.

Reply:
This seems to be unused leftover code from _that other repo of yours_. I used it during testing here and there, I'll remove it.